011-php-code-formatting.md

```diff
@@ -0,0 +1,94 @@
+
+However, once flagged in a pull request, to automatically fix these errors, developers can just run the `composer format` command in Homestead and then commit and push up their code, confident that the errors were addressed and the style checks should pass.
+
+## Drawbacks
+
+### Two-Worlds Problem
+
+In an ideal world, the PHP CS Fixer tool would run after Prettier and both be triggered by the pre-commit hook. However, based on our development environment setup using Homestead, we typically run `composer` commands within the vagrant virtual machine and `npm` commands outside the vagrant box (watching and compiling assets with via `npm` tool has proven to run rather slow on the VM and thus we typically run them within our local environment).
+
+Due to this setup, one drawback with the PHPCS Fixer tool is that the tool cannot be triggered as a pre-commit hook to run when Prettier is run.
+
+This [post](https://sebastiandedeyne.com/running-php-cs-fixer-on-every-commit-with-husky-and-lint-staged/) discusses setting up both Prettier and PHP CS Fixer to run using the [`lint-staged`](https://github.com/okonet/lint-staged) NPM package, which allows configuring use of code quality tools alongside Prettier. However, I think it only works because they are both running within the same environment (and not split between a VM and local computer).
+
+While it is unfortunate it is not a deal breaker and as described in the "PHP Code Standards Fixer" section above, the checks will happen via the CI runs and devs can fix errors locally by running `composer format`, commit and push them up.
+
+### Contradictions
+
+Another potential drawback is the possibility of contradicting fixes. Where depending on the order of operations, a style fix could keep getting edited back and forth between Prettier and PHP CS Fixer.
+
+This is theoretical, since I have not encountered this, but I think it will be something to look out for initially if we move forward with this RFC.
+
+Luckily, both Prettier and the PHP CS Fixer rulesets adhere to the same PSRs so I am hopeful that this will not really be an issue.
+
+For a quick visual comparison, here is the same file edited separately by the two tools:
+
+#### Prettier Changes
+
+You can see in these [Prettier file changes](https://github.com/DoSomething/rogue/pull/1094/files#diff-5c9547d315e53dd95227ca9984b56d06), it edited the line-lengths of the code.
+
+#### PHP CS Fixer Changes
+
+You can see in these [PHP CS Fixer file changes](https://github.com/DoSomething/rogue/pull/1092/files#diff-5c9547d315e53dd95227ca9984b56d06), it sorted the class imports to be alphabetical and removed the spacing around string concatenation. This would be an example where we would want our PHP CS Fixer ruleset to not fix string concatenation, since Prettier styles them with a space around string concatenation.
```

Comment:
I'm confused about this case - if I run `composer format` to fix the ❌ on my PR, won't that spacing change it made automatically be reverted when I save the files because Prettier will run on save?

Reply:
Ya, I mention this in the "Contradictions seciton:

> Another potential drawback is the possibility of contradicting fixes. Where depending on the order of operations, a style fix could keep getting edited back and forth between Prettier and PHP CS Fixer.

But I [discuss this](https://github.com/DoSomething/rfcs/pull/11#discussion_r469318473) and propose a way to address once implementing. Aiming to remove the rules in PHP CS Fixer that conflict with Prettier so that we cater to each of their strengths.